apps/condo/domains/meter/tasks/sendVerificationDateReminder.js

```diff
@@ -0,0 +1,245 @@
+                const message = sentMessages.find(sentMessage => {
+                    // we have to check the following things
+                    // 1. Get message for particular resident
+                    // 2. Filter out messages related to other meters
+                    // 3. Filter out messages sent before current reminder window
+                    const messageCreateAt = dayjs(sentMessage.createdAt)
+                    const endOfReminderWindow = dayjs(meter.nextVerificationDate)
+                    const startOfReminderWindow = endOfReminderWindow.add(-reminderWindowSize, 'day')
+
+                    return sentMessage.user.id === resident.user.id
+                        && meter.id === sentMessage.meta.data.meterId
+                        && messageCreateAt.unix() >= startOfReminderWindow.unix()
+                })
+                return message == null
+            })
+
+            // create a pair again
+            // and filter out meter with empty residents
+            return {
+                meter,
+                residents: residentsWithoutNotifications,
+            }
+        })
+        .filter(item => item.residents.length > 0)
+}
+
+const getOrganizationLang = async (context, id) => {
+    const organization = await Organization.getOne(context, { id, deletedAt: null })
+
+    /**
+     * Detect message language
+     * Use DEFAULT_LOCALE if organization.country is unknown
+     * (not defined within @condo/domains/common/constants/countries)
+     */
+    return get(COUNTRIES, get(organization, 'country.locale'), DEFAULT_LOCALE)
+}
+const generateReminderMessages = async ({ context, reminderWindowSize, reminders }) => {
+    const messages = []
+
+    await Promise.all(reminders.map(async (reminder) => {
+        const { meter, residents } = reminder
+        const lang = await getOrganizationLang(context, meter.organization.id)
+
+        // prepare a message for each resident
+        messages.push(
+            ...residents.map(resident => ({
+                sender: { dv: 1, fingerprint: 'meters-validation-date-cron-push' },
+                to: { user: { id: resident.user.id } },
+                type: METER_VERIFICATION_DATE_REMINDER_TYPE,
+                lang,
+                meta: {
+                    dv: 1,
+                    data: {
+                        reminderDate: dayjs(meter.nextVerificationDate).locale(lang).format('D MMM'),
+                        meterId: meter.id,
+                        userId: resident.user.id,
+                        residentId: resident.id,
+                    },
+                },
+            }))
+        )
+    }))
+
+    return messages
+}
+
+const sendReminders = async ({ context, messages }) => {
+    await Promise.all(messages.map(async (message) => {
+        await sendMessage(context, message)
+    }))
+}
+
+const sendVerificationDateReminder = async ({ date, searchWindowDaysShift, daysCount }) => {
```

Comment:
Not sure, since we would like to operate not only with date range.  Just look at readMetersPage implementation. the searchWindowDaysShift required to separate 30 days and 60 days reminders